common/tests/unit/test_CipherEngine.cpp

```diff
@@ -0,0 +1,148 @@
+#define BOOST_TEST_MAIN
+
+#define BOOST_TEST_MODULE keyEncryptionDecryption
+#include <boost/filesystem.hpp>
+#include <boost/test/unit_test.hpp>
+//Local private includes
+#include "../../include/common/CipherEngine.hpp"
+#include "../../include/common/Util.hpp"
+
+#include "common/libjson.hpp"
+
+//Standard includes
+#include <string.h>
+#include <iostream>
+#include <fstream>
+
+using namespace std;
+using namespace SDMS;
+
+BOOST_AUTO_TEST_SUITE(KeyEncryptionDecryptionTest)
+
+BOOST_AUTO_TEST_CASE(test_EncryptionDecryption)
+{
+
+    std::cout << "THIS IS A NEW FLAG" << std::endl;
+    unsigned char key[32];
+    //CipherEngine::generateEncryptionKey(key);
+    readFile("../../build/core/server/datafed-token-key.txt", 32, key);
+    
+    //Construct
+    CipherEngine encryptCipher(key);
+
+    //Sets struct CipherString: which contains cipherText, cipherIV, cipherPaddedLen
+    string msg = "1234567890yoDa56Bx5yobvJYEjdGr2YpGYJybE7x4Bq42pQ3zuXCb8YQyn0EqEB7vjPx3GlNlKwkEsMn1234567890";
+
+    //string msg = "Hello World";
+    //CipherEngine::generateIV(iv);
+
+    //Here if parties would like to use their own IV GENERATOR
+    //CipherEngine::CipherString returnObj = testCipher.encrypt(iv, msg);
+    
+    CipherEngine::CipherString returnObj2 = encryptCipher.encrypt(msg);
+ 
```

Comment:
change name 'returnObj2' is not a great choice. Maybe something like encoded_encrypted_packet or something like that.